File: src/Python/501-600/590.N-aryTreePostorder.py

class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution(object):
    def trace(self, node, nodelist):
        if node is None:
            return
        for n in node.children:
            self.trace(n, nodelist)
        nodelist.append(node.val)

    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        nodelist = []
        self.trace(root, nodelist)
        return nodelist

if __name__ == '__main__':
    solution = Solution()
    root = Node(1, [])
    root.children.append(Node(3, []))
    root.children.append(Node(2, []))
    root.children.append(Node(4, []))
    print(solution.postorder(root))
else:
    pass

File: src/CSharp/501-600/590.N-aryTreePostorder.cs

using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.NAryTreePostorder
{

    // Definition for a Node.
    public class Node
    {
        public int val;
        public IList<Node> children;

        public Node() { }
        public Node(int _val, IList<Node> _children)
        {
            val = _val;
            children = _children;
        }
    }

    public class Solution
    {
        List<int> list;

        void Trace(Node node)
        {
            if (node == null)
                return;
            list.Add(node.val);
            foreach (var n in node.children)
            {
                Trace(n);
            }
        }

        public IList<int> Posteorder(Node root)
        {
            list = new List<int>();
            Trace(root);
            return list;
        }
    }
}

